src/frontend/src/handler/create_source.rs

```diff
@@ -1364,9 +1379,25 @@ pub async fn bind_create_source(
         session.get_database_and_schema_id_for_create(schema_name.clone())?;
 
     if !is_create_source && with_properties.is_iceberg_connector() {
-        return Err(
-            ErrorCode::BindError("can't create table with iceberg connector".to_string()).into(),
-        );
+        return Err(ErrorCode::BindError(
+            "can't CREATE TABLE with iceberg connector\n\nHint: use CREATE SOURCE instead"
+                .to_string(),
+        )
+        .into());
+    }
+    if is_create_source {
+        match source_schema.format {
+            Format::Upsert => {
+                return Err(ErrorCode::BindError(format!(
+                    "can't CREATE SOURCE with FORMAT UPSERT\n\nHint: use CREATE TABLE instead\n\n{}",
+                    hint_upsert(&source_schema.row_encode)
+                ))
+                .into());
+            }
+            _ => {
+                // TODO: enhance error message for other formats
```

Comment:
Debezium (DebeziumMongo), Canal, Maxwell all require `table` rather than `source`.

Reply:
Yeah, I thought only `PLAIN` is for `SOURCE`. Now I'm not sure when I found none/native. 🤪

Reply:
`native` is for datagen, so it should work in both source and table. Not sure about `None` 😇

Reply:
Link to the original PR: https://github.com/risingwavelabs/risingwave/pull/12306

TL;DR: Keywords like `NATIVE` will never be explicitly written by users in their SQL, but appear after normalization and persistence. When we're going to do schema change, we have to reparse the SQL (under current implementation) so `NATIVE` must be accepted as a keyword.